### ISSUE TITLE
Remove automatic dasherization of modifiers

### ISSUE THREAD

**Issue by rwjblue** [ember-cli/ember-resolver#506]
We should add `modifier` guard to this conditional:

https://github.com/ember-cli/ember-resolver/blob/59d6e19cfd19c8f16f6f1cada771112da6231b9a/addon/resolvers/classic/index.js#L179-L186

This would prevent folks from falling prey to the kinds of issues that triggered https://github.com/ember-template-lint/ember-template-lint/pull/1078.

Tasks:

* Make PR for:
  * Add `modifier` guard there
  * Add a test showing that `resolve('modifier:didInsert')` does not find `app-name/modifiers/did-insert`
* Land that PR
* Release with major bump

If we actually think folks are hitting this, we _could_ add a deprecation first within the current major version then remove and release a new major bump, but TBH I'm not sure that is necessarily warranted.

**Comment by Turbo87** [comment]
I'm not sure I fully understand this. Are you saying the file should be didInsert.js?

**Comment by rwjblue** [comment]
@Turbo87 - I'm saying that for template invocable things we do _not_ want to convert `type:fooBar` into `type:foo-bar` before looking up, and that having `app/modifiers/fooBar.js` on disk is "fine". Automatically converting `{{fooBar}}` to `{{foo-bar}}` **will** cause issues (this means that you cannot use a local property named `fooBar` if you also happen to have an invokable with that name), and as of when [emberjs/rfcs#432](https://github.com/emberjs/rfcs/blob/master/text/0432-contextual-helpers.md) is fully implemented you _will_ be able to have modifiers as local properties (meaning `<div {{fooBar}}></div>` _could_ refer to `this.fooBar` with the implicit this fallback).

For background, we have had a number of fixes to deal with these normalization issues in template invoked things:

* https://github.com/ember-cli/ember-resolver/pull/167
* https://github.com/emberjs/ember.js/issues/18215
* https://github.com/ember-cli/ember-resolver/pull/392
* https://github.com/ember-cli/ember-resolver/pull/396

tldr; when we added `modifier` to the list of things that could be provided by apps, we should have also excluded it from the normalization (like the other template invokable types).

**Comment by Turbo87** [comment]
hmm, I would consider this a breaking change though. the template lint rule was essentially triggered by one of our clients mixing these invocations styles and we wanted to bring some consistency to it. in other words: there are definitely apps out there that rely on this automatic dasherization.

**Comment by rwjblue** [comment]
> hmm, I would consider this a breaking change though

Indeed, see steps above indicating major version bump 😉
